### PR TITLE
TriggerAuthentication/Vault: add support for vault namespace

### PR DESCRIPTION
**Pull request by chapurlatn**
Signed-off-by: Nicolas Chapurlat <nc@coorganix.com>

Related to this issue: [#2084](https://github.com/kedacore/keda/issues/2084)
Related to this discussion: [#2080](https://github.com/kedacore/keda/discussions/2080)

### Changes

- Add an optional "namespace" in Keda TriggerAuthentication spec: `spec.hashiCorpVault.namespace`
- Optionally configure the Vault namespace in the Vault client
- **REMOVED FROM THIS PR**: Make sure the VAULT_NAMESPACE env var never takes precedence (Vault client default behaviour)

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added **no test yet in vault section (I performed a local integration test I could provide)**
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*  => **not applicable**
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*  => **Done: [https://github.com/kedacore/keda-docs/pull/522](https://github.com/kedacore/keda-docs/pull/522)**
- [x] Changelog has been updated

Relates to https://github.com/kedacore/keda/issues/2084

### CODE REVIEW

File: pkg/scaling/resolver/hashicorpvault_handler.go [hunk 1]
```diff
@@ -28,6 +29,16 @@ func NewHashicorpVaultHandler(v *kedav1alpha1.HashiCorpVault) *HashicorpVaultHan
 // Initialize the Vault client
 func (vh *HashicorpVaultHandler) Initialize(logger logr.Logger) error {
 	config := vaultApi.DefaultConfig()
+
+	// do not take into account vault namespace env variable
+	// It prevents any conflict with env during testing, just use HashiCorpVault inputs.
+	// Vault client creation use env var to override any conf (NewClient)
+	// We should probably also unset EnvVaultToken for the same purpose.
+	err := os.Unsetenv(vaultApi.EnvVaultNamespace)
```
Comment (ahmelsayed) [inline pkg/scaling/resolver/hashicorpvault_handler.go:37]: Is this something that would be set automatically on all pods including KEDA, or is it something the user is explicitly setting?

Comment (chapurlatn) [reply pkg/scaling/resolver/hashicorpvault_handler.go:37]: Vault handles several way to provide token and namespace with a precedence (explicit overriden by file and overriden by env var).
I'm working also on Kubernetes appliance and we've integrated Vault Enterprise. We've noticed that when using explicit namespace and/or token configuration (from a Kube spec in our example), it's really painful for developers when testing. They may have configured some Vault namespace locally using an env var and it overrides the conf in the tested soft.
It's probably a bad idea I had to also push that code in that PR, I propose to remove that code and eventually, If you want to, provide another PR to ignore both VAULT_NAMESPACE and VAULT_TOKEN env variable.

**Comment by chapurlatn** [comment]
> Is this related with the issue [TriggerAuthentication/Vault: add support for vault namespace](https://github.com/kedacore/keda/issues/2084)?

**Comment by chapurlatn** [comment]
> Is this related with the issue [TriggerAuthentication/Vault: add support for vault namespace](https://github.com/kedacore/keda/issues/2084)?

Yes @JorTurFer, I added a ref in the main comment.

**Review by zroubalik — APPROVED** [review]

**Comment by zroubalik** [comment]
Thanks for the contribution!

**Pull request merged**